src/LogFileReader.cpp

```diff
@@ -75,11 +77,23 @@ bool LogFileReader::readAll(const QString& fileName)
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     // TODO implement this first
+QStringList sections= line.split(", ");
+if(sections.length()!=3)
+{
+    return false;
+}
 
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
+QString timeString= sections.at(0);
+//Converts timeString to time for battery data
+batteryData.time= QTime::fromString(timeString,"hh:mm:ss.zzz");
```

Comment:
This file is very clean, well done. I would note that in general writing strings in non print functions, as well as straight numbers in functions calls is poor structure. Normally, you would either derive such a value from somewhere else, or you would give it a descriptive name as a constant variable somewhere.

In this case, the "hh:mm:ss.zzz" constant, since it can't be derived elsewhere, it should made as a constant. Normally how we do constants in C++ on the solar car team is adding a namespace at the top. So the code would be something like:

namespace 
{
    const QString SOME_DESCRIPTIVE_NAME = "hh:mm:ss.zzz";
}
I would suggest something similar for the line QStringList splitLine = line.split(", "); as well as the 3.